github_handler.py

```
@@ -0,0 +1,358 @@
+                    logger.warning(f"Access forbidden to repository: {owner}/{repo}")
+                    return None
+                elif response.status != 200:
+                    logger.error(f"GitHub API returned status {response.status} for {owner}/{repo}")
+                    return None
+                
+                data = await response.json()
+                logger.info(f"Successfully fetched metadata for {owner}/{repo}")
+                return data
+                
+    except Exception as e:
+        logger.error(f"Error fetching repo metadata for {owner}/{repo}: {str(e)}", exc_info=True)
```

Ruff (0.14.3)

Use explicit conversion flag

Replace with conversion flag

(RUF010)